pkg/ansible/generate.go

```diff
@@ -0,0 +1,257 @@
+# openshift_aws_users
+openshift_aws_ssh_key_name: libra # myuser_key
+
+# This will ensure these user and public keys are created.
+#openshift_aws_users:
+#- key_name: myuser_key
+#  username: myuser
+#  pub_key: |
+#         ssh-rsa AAAA
+
+# When building the AMI, specify the user to ssh to the instance as.
+# openshift_aws_build_ami_ssh_user: root
+
+# --------- #
+# AMI Build #
+# --------- #
+# Variables in this section apply to building a node AMI for use in your
+# openshift cluster.
+
+# must specify a base_ami when building an AMI
+openshift_aws_base_ami: ami-ae7bfdb8 # ami-12345678
+
+# this is the ami we built
+openshift_aws_ami: ami-9a0d95e0 # ami-2f53ce55
+
+# when creating an encrypted AMI please specify use_encryption
+#openshift_aws_ami_encrypt: False
+
+# -- #
+# S3 #
+# -- #
+
+# Create an s3 bucket.
+openshift_aws_create_s3: false
+
+# --- #
+# ELB #
+# --- #
+
+# openshift_aws_elb_name will be the base-name of the ELBs.
+# TODO: looks like this is supposed to just be basename variant
+#openshift_aws_elb_name: dgoodwin-dev
+
+# custom certificates are required for the ELB
+openshift_aws_iam_cert_path: /home/abutcher/rhat/certs/wildcard-flibberty-jibbet.com.crt  # '/path/to/wildcard.<clusterid>.example.com.crt'
+openshift_aws_iam_cert_key_path: /home/abutcher/rhat/certs/wildcard-flibberty-jibbet.com.key  # '/path/to/wildcard.<clusterid>.example.com.key'
+openshift_aws_iam_cert_chain_path: /home/abutcher/rhat/certs/ca.crt  # '/path/to/cert.ca.crt'
+
+openshift_aws_create_iam_role: True
+
+{{ range $key, $value := . }}
+   {{ $key }}: {{ $value }}
+{{ end }}
+`
+
+// TODO: these {{}} don't play nice in a go template, complete this when we get to processing
+// node groups dynamically
+/*
+# tags
+openshift_aws_master_group_tags:
+  Name: "{{ openshift_aws_clusterid }}-master"
+  host-type: master
+  sub-host-type: default
+
+openshift_aws_node_group_compute_tags:
+  Name: "{{ openshift_aws_clusterid }}-compute"
+  host-type: node
+  sub-host-type: compute
+
+openshift_aws_node_group_infra_tags:
+  Name: "{{ openshift_aws_clusterid }}-infra"
+  host-type: node
+  sub-host-type: infra
+*/
+
+const (
+	cloudProviderKindVar = "openshift_cloudprovider_kind"
+	awsClusterIDVar      = "openshift_aws_clusterid"
+	elbBasenameVar       = "openshift_aws_elb_basename"
+	awsVPCNameVar        = "openshift_aws_vpc_name"
+	awsRegionVar         = "openshift_aws_region"
+	awsBaseAMINameVar    = "openshift_aws_base_ami_name"
+)
+
+type VarsGenerator struct {
+	Vars map[string]string
+}
+
+func NewVarsGenerator(cluster *coapi.Cluster) *VarsGenerator {
+
+	// Any vars that change from cluster to cluster need to be defined here:
+	inv := &VarsGenerator{
+		Vars: map[string]string{
+			cloudProviderKindVar: "aws",
+			awsRegionVar:         "us-east-1",
+			awsClusterIDVar:      cluster.Name,
+			elbBasenameVar:       cluster.Name,
+			awsVPCNameVar:        cluster.Name,
+			awsBaseAMINameVar:    fmt.Sprintf("%s-ami-base", cluster.Name),
```

Comment:
I would have thought that you'd specify the `openshift_aws_ami` var and not the `openshift_aws_base_ami_name`